backend/compact-connect/lambdas/license-data/handlers/utils.py

```diff
@@ -123,3 +127,39 @@ def authorized(event: dict, context: LambdaContext):
                 return {'statusCode': 403}
             return fn(event, context)
         return authorized
+
+
+def sqs_handler(fn: Callable):
+    """
+    Process messages from the ingest queue.
+
+    This handler uses batch item failure reporting:
+    https://docs.aws.amazon.com/lambda/latest/dg/example_serverless_SQS_Lambda_batch_item_failures_section.html
+    This allows the queue to continue to scale under load, even if a number of the messages are failing. It
+    also improves efficiency, as we don't have to throw away the entire batch for a single failure.
+    """
+    @wraps(fn)
+    @logger.inject_lambda_context
+    def process_messages(event, context: LambdaContext):  # pylint: disable=unused-argument
```

Comment:
`pylint` will identify whenever there is an unused function argument, which helps to ensure that you aren't defining function signatures that are unnecessarily big. A very common case like this, though, is that you are implementing an interface function that provides arguments that you simply don't need, but it's part of the interface, so you have to include it. Lambda handler functions always have to have this `context` argument, even though it is usually not needed. It's common enough that I tend to just disable the rule and move on, so I can avoid adding noisy comments for probably the most common disable line. I'm happy to err on the side of verbose comments if that's what you'd prefer.

Reply:
Got it. Fine with leaving as-is then. Does lambda complain if you leave out `context`? I'm used to python being pretty flexible when it comes to stuff like that.

Reply:
Yeah. Python will be very flexible if the function you are calling has optional arguments that you don't provide, but if you provide an argument that isn't listed, it will always fail (unless you put `*args`, or `**kwargs`, in which case, we've still got an unused argument here). Lambda will provide two positional arguments to the entry-point to a lambda, so it will fail if we don't accept them.